### PR TITLE
[Fix] Remove --no-optional when running npm

### PR DESCRIPTION
**Pull request by henrytao-me**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/app-extension-libs/issues/1112

This PR fixes `shopify serve` issue when running Argo extension. `chokidar ` is an optional dependency that webpack uses for Argo. Considering Shopify cli is for development, it makes sense to not use `--no-optional` option.

### WHAT is this pull request doing?

- Remove `--no-optional` and fix test.

### Update checklist

- [x] I've added a CHANGELOG entry for this PR

### How to 🎩 
- Do 
```
dev cd shopify-app-cli
./bin/shopify create extension
```
- Follow the steps to create new Argo Admin Extension
- Go to new extension directory and point to correct local shopify-app-cli
```
../bin/shopify-app-cli/bin/shopify serve
```
- Verify that extension is served correctly without having to do `npm install` again.

### CODE REVIEW

File: lib/shopify-cli/js_deps.rb
Comment (andyw8) [inline lib/shopify-cli/js_deps.rb:62]: @tanema looks like you added this in https://github.com/Shopify/shopify-app-cli/pull/517 – just checking if you can see any problem with removing it.

File: CHANGELOG.md
Comment (andyw8) [inline CHANGELOG.md:3]: @henrytao-me can you please include the PR #

Comment (henrytao-me) [reply CHANGELOG.md]: Added! 🙇

File: CHANGELOG.md
Comment (fredriculrich) [inline CHANGELOG.md:3]: I think it would make sense to update this text. You haven't changed it just for extensions but for all projects in the CLI with NPM dependencies.

Comment (henrytao-me) [reply CHANGELOG.md]: Updated!

**Comment by fredriculrich** [comment]
From an extensions project standpoint this LGTM. I'd wait for an approval from the CLI team to make sure this doesn't have any effects for them that I am unaware of.

**Review by fredriculrich — APPROVED** [review]

**Review by andyw8 — APPROVED** [review]

**Pull request merged**
